src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle side must be greater than zero');
+    }
 
-export class Circle {
+    if (a + b <= c || a + c <= b || c + b <= b) {
+      throw new Error('It`s not a triangle');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter: number = 0.5 * (this.a + this.b + this.c);
+
+    return Number(Math.sqrt(semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c)).toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than zero');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?